### ISSUE TITLE
typedef struct name1 name2

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#430]
C:
```c
int main(){
        typedef struct tnode TNODE;
        struct tnode {
                int count;
                TNODE *left, *right;
        };
        TNODE s, *sp;
        (void)(s);
        (void)(sp);
        return 0;
}
```

```
`-FunctionDecl 0x3e39770 <7.c:1:1, line:11:1> line:1:5 main 'int ()'
  `-CompoundStmt 0x3e8c5f8 <col:11, line:11:1>
    |-DeclStmt 0x3e8c0d8 <line:2:2, col:28>
    | |-RecordDecl 0x3e8bf70 <col:10, col:17> col:17 struct tnode
    | `-TypedefDecl 0x3e8c070 <col:2, col:23> col:23 referenced TNODE 'struct tnode':'struct tnode'
    |   `-ElaboratedType 0x3e8c020 'struct tnode' sugar
    |     `-RecordType 0x3e8c000 'struct tnode'
    |       `-Record 0x3e8c0f0 'tnode'
    |-DeclStmt 0x3e8c308 <line:3:2, line:6:3>
    | `-RecordDecl 0x3e8c0f0 prev 0x3e8bf70 <line:3:2, line:6:2> line:3:9 struct tnode definition
    |   |-FieldDecl 0x3e8c190 <line:4:3, col:7> col:7 count 'int'
    |   |-FieldDecl 0x3e8c268 <line:5:3, col:10> col:10 left 'TNODE *'
    |   `-FieldDecl 0x3e8c2c0 <col:3, col:17> col:17 right 'TNODE *'
    |-DeclStmt 0x3e8c418 <line:7:2, col:14>
    | |-VarDecl 0x3e8c330 <col:2, col:8> col:8 used s 'TNODE':'struct tnode'
    | `-VarDecl 0x3e8c3a0 <col:2, col:12> col:12 used sp 'TNODE *'
    |-CStyleCastExpr 0x3e8c4d0 <line:8:2, col:10> 'void' <ToVoid>
    | `-ImplicitCastExpr 0x3e8c4b8 <col:8, col:10> 'TNODE':'struct tnode' <LValueToRValue>
    |   `-ParenExpr 0x3e8c498 <col:8, col:10> 'TNODE':'struct tnode' lvalue
    |     `-DeclRefExpr 0x3e8c430 <col:9> 'TNODE':'struct tnode' lvalue Var 0x3e8c330 's' 'TNODE':'struct tnode'
    |-CStyleCastExpr 0x3e8c598 <line:9:2, col:11> 'void' <ToVoid>
    | `-ImplicitCastExpr 0x3e8c580 <col:8, col:11> 'TNODE *' <LValueToRValue>
    |   `-ParenExpr 0x3e8c560 <col:8, col:11> 'TNODE *' lvalue
    |     `-DeclRefExpr 0x3e8c4f8 <col:9> 'TNODE *' lvalue Var 0x3e8c3a0 'sp' 'TNODE *'
    `-ReturnStmt 0x3e8c5e0 <line:10:2, col:9>
      `-IntegerLiteral 0x3e8c5c0 <col:9> 'int' 0

```
Go:
```go
package main

func main() {
        __init()
        type tnode struct {
        }
        type TNODE struct {
        }
        var s TNODE
        var sp []TNODE
        (s)
        (sp)
        return
}
func __init() {
}
```
Body of struct are lost.

**Comment by Konstantin8105** [comment]
```c
	diag("name of struct inside struct")
	{
		typedef struct TI TI;
		struct TI{
			TI *left, *right;
			double varTI;
		};
		TI t1;
		t1.varTI = 4.3;
		TI t2;
		t2.varTI = 4.1;
		TI tt;
		tt.left       = &t1;
		(*tt.left).right = &t2;
		tt.right      = &t2;
		is_eq((*tt.left ).varTI, 4.3);
		is_eq((*(*tt.left).right).varTI, 4.1);
		is_eq((*tt.right).varTI, 4.1);
	}
```
```go
	noarch.Printf([]byte("# \x00"))
	noarch.Printf([]byte("name of struct inside struct\x00"))
	noarch.Printf([]byte("\n\x00"))
	{
		type TI struct {
			left  []TI
			right []TI
			varTI float64
		}
		var t1 TI
		t1.varTI = 4.3
		var t2 TI
		t2.varTI = 4.1
		var tt TI
		tt.left = (*[1]TI)(unsafe.Pointer(&t1))[:]
		(tt[noarch.InterfaceToInt(0 .left)]).right = (*[1]TI)(unsafe.Pointer(&t2))[:]
		tt.right = (*[1]TI)(unsafe.Pointer(&t2))[:]
		if approxf(((tt[noarch.InterfaceToInt(0 .left)]).varTI), (4.3), func() int { /// <<-- warning here
...
		if approxf((((tt[noarch.InterfaceToInt(0 .left)])[noarch.InterfaceToInt((0).right)]).varTI), (4.1), func() int { /// <<-- warning here
...
```
```
    		build/tests/struct/main_test.go:1440: undefined: noarch.InterfaceToInt
    		build/tests/struct/main_test.go:1440: 0.left undefined (type int has no field or method left)
    		build/tests/struct/main_test.go:1442: undefined: noarch.InterfaceToInt
    		build/tests/struct/main_test.go:1442: 0.left undefined (type int has no field or method left)
    		build/tests/struct/main_test.go:1470: undefined: noarch.InterfaceToInt
    		build/tests/struct/main_test.go:1470: 0.left undefined (type int has no field or method left)
    		build/tests/struct/main_test.go:1475: undefined: noarch.InterfaceToInt
    		build/tests/struct/main_test.go:1475: 0.left undefined (type int has no field or method left)
    		build/tests/struct/main_test.go:1475: undefined: noarch.InterfaceToInt
    		build/tests/struct/main_test.go:1475: 0.right undefined (type int has no field or method right)
    		build/tests/struct/main_test.go:1475: too many errors
```

**Comment by Konstantin8105** [comment]
Closed #460
